07week/ticTacToe/script.js

```diff
@@ -1,31 +1,111 @@
 'use strict';
 
+class Square extends React.Component{
+  render() {
+    return (
+   <button className="square" onClick={() => this.props.onClick()}>{this.props.value}</button>
+    );
+  }
+}
+
 class TicTacToe extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+                     //if in state no need for this.something...Player turn goes in the state because the state is changeing
+      // players1: null,
+      squares: [
+        "", "", "", "", "", "", "", "", ""
+      ],
+     xNext: true,
+    };
   }
 
+  handleClick(i){
+
+    const squares = this.state.squares.slice();
+    if(calculateWin(squares) || squares[i]) {
+        return;
+    }
+    squares[i] = this.state.xNext ? 'X' : 'O';
+    this.setState({
+      squares: squares,
+      xNext: !this.state.xNext,
+    })
+    // this.playerTurn = this.playerTurn === 'X' ? 'O' : 'X'
+    // return div[data-cell] = playerTurn
+    // console.log(this.renderSquare());
+  }
+  renderSquare(i) {
+    return (
+          <Square
```

Comment:
Very nice use of a render function, but remember to code DRY (don't repeat yourself).
I'd recommend instead making this a `renderSquares()` function that maps through `this.state.squares`. This way you have access to the index of each square which you can pass into `this.state.squares[index]` as well as `this.handleClick(index)`.

Reply:
Also you've got some funky indentation here. coughCOPYPASTINGFROMTHEREACTDOCScough